docs(understanding-airbyte): Add file sync and permission sync documentation (do not merge) (#55783)
airbytehq/airbyte: docs/understanding-airbyte/permission-sync.md

# Airbyte Permission Sync

Permission Sync is a capability in Airbyte that allows you to transfer access control information and permission structures between systems. This document explains how Permission Sync works, which connectors support it, and how to use it.

## Overview

When transferring data between systems, it's often important to maintain not just the data itself but also the permission structures that govern access to that data. Permission Sync addresses this need by:

> - Preserving user and group access controls
> - Maintaining role-based permissions
> - Transferring ownership information
> - Replicating sharing settings

Reviewer: ```suggestion
- Preserving user and group access controls.
- Maintaining role-based permissions.
- Transferring ownership information.
- Replicating sharing settings.
```

This ensures that when data is moved between systems, the appropriate access controls are maintained.

## How Permission Sync Works

When using Permission Sync:

> 1. The source connector extracts both data and associated permission metadata
> 2. Permission structures are mapped between source and destination systems
> 3. The destination connector applies compatible permission settings
> 4. User and group mappings are maintained where possible

Reviewer: ```suggestion
1. The source connector extracts both data and associated permission metadata.
2. Permission structures are mapped between source and destination systems.
3. The destination connector applies compatible permission settings.
4. User and group mappings are maintained where possible.
```

Permission Sync can work alongside regular data synchronization or File Sync operations.

## Supported Connectors

Permission Sync is currently in early development with limited connector support. The following connectors are planned to support Permission Sync:

> ### Sources
> - Microsoft SharePoint (in development)

Reviewer: There should always be a blank line after a header. Also before lists.

```suggestion
### Sources

- Microsoft SharePoint (in development)
```

# Airbyte Permission Sync

Permission Sync is a capability in Airbyte that allows you to transfer access control information and permission structures between systems. This document explains how Permission Sync works, which connectors support it, and how to use it.

## Overview

When transferring data between systems, it's often important to maintain not just the data itself but also the permission structures that govern access to that data. Permission Sync addresses this need by:

- Preserving user and group access controls.
- Maintaining role-based permissions.
- Transferring ownership information.
- Replicating sharing settings.

This ensures that when data is moved between systems, the appropriate access controls are maintained.

## How Permission Sync Works

When using Permission Sync:

1. The source connector extracts both data and associated permission metadata.

> 2. Permission structures are mapped between source and destination systems.
> 3. The destination connector applies compatible permission settings.
> 4. User and group mappings are maintained where possible.
> 
> Permission Sync can work alongside regular data synchronization or File Sync operations.

Reviewer: ```suggestion
2. Permission structures are replicated from the source and sent as records.
3. The destination connector receives permission information as incoming records.
4. Permission logic allows restrictions to be reconstructed in downstream applications.

While Permission Sync and File Sync connections can often complement each other, they are distinct and separate features and should be set up as separate connnections.
```

## Supported Connectors

Permission Sync is currently in early development with limited connector support. The following connectors are planned to support Permission Sync:

### Sources

- Microsoft SharePoint (in development)
- Google Drive (planned)
- Box (planned)

### Destinations

> - S3 (in development)
> - Google Cloud Storage (planned)
> - Azure Blob Storage (planned)

Reviewer: Update to reflect the following info:

Permission sync uses standard record-type processing so it is compatible with all Airbyte destinations. 

```suggestion
```
